lib/secp256k1/secp256k1.ex

```diff
@@ -235,6 +187,30 @@ defmodule Bitcoinex.Secp256k1 do
     end
   end
 
+  @doc """
+    force_even_y returns the negated private key
+    if the associated Point has an odd y. Otherwise
+    it returns the private key
+  """
+  @spec force_even_y(PrivateKey.t()) :: PrivateKey.t() | {:error, String.t()}
+  def force_even_y(%PrivateKey{} = privkey) do
+    case PrivateKey.to_point(privkey) do
+      {:error, msg} ->
+        {:error, msg}
+
+      pubkey ->
+        if Point.is_inf(pubkey) do
```

Comment:
this error cause will never be returned(as it will continue execution)

if Point.is_inf(pubkey) do
 {:error, "pubkey is infinity. bad luck"}
else
     if Point.has_even_y(pubkey) do
          privkey
        else
          %PrivateKey{d: Params.curve().n - privkey.d}
        end
end